### DISCUSSION TITLE
How to define **{Create,Delete,Update}Args?

### DISCUSSION THREAD

**Discussion by e-jigsaw** [blitz-js/blitz#516] [General] [answered]
`blitz generate all *` generates `import db, { *CreateArgs } from "db"` to `mutations/create*.ts`. But, `*CreateArgs`not defined in `db/index.ts`.

It means, should I write `*Args` type each times?

Thanks

**Comment by flybayer** [comment, marked as answer]
Good question!

**Short answer:** you don't have to define any of those types. Those types are automatically defined by Prisma.

**Long answer:** The contents of `db.ts` is:

```ts
import {PrismaClient} from '@prisma/client'
export * from '@prisma/client'

const prisma = new PrismaClient()

export default prisma
```

Notice the `export * from '@prisma/client'` line. That line exports all of types defined by Prisma, including the *{Create,Delete,Update}Args types.

You can see all the Prisma type definitions by opening `node_modules/.prisma/client/index.d.ts`

**Reply by e-jigsaw** [reply]
Thank you for support.

I read `node_modules/@prisma/client/index.d.ts`. It just export from `.prisma/client`. I thought "where is `.prisma` ...?". soon I found `node_modules/.prisma/client` and `node_modules/.prisma/client/index.d.ts`, it defined whole types. wow

**Reply by flybayer** [reply]
Oh yes, that path changed in prisma beta.4. I updated my answer!
